packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit_api_impls.dart

```diff
@@ -381,6 +420,53 @@ class WKNavigationDelegateHostApiImpl extends WKNavigationDelegateHostApi {
       await create(instanceId);
     }
   }
+
+  /// Calls [setDidFinishNavigation] with the ids of the provided object instances.
+  Future<void> setDidFinishNavigationFromInstance(
+    WKNavigationDelegate instance,
+    void Function(WKWebView, String?)? didFinishNavigation,
+  ) {
+    int? functionInstanceId;
+    if (didFinishNavigation != null) {
+      functionInstanceId = instanceManager.getInstanceId(didFinishNavigation);
+      functionInstanceId ??=
+          instanceManager.tryAddInstance(didFinishNavigation);
+    }
+    return setDidFinishNavigation(
+      instanceManager.getInstanceId(instance)!,
+      functionInstanceId,
+    );
+  }
+}
+
+/// Flutter api implementation for [WKNavigationDelegate].
+class WKNavigationDelegateFlutterApiImpl
+    extends WKNavigationDelegateFlutterApi {
+  /// Constructs a [WKNavigationDelegateFlutterApiImpl].
+  WKNavigationDelegateFlutterApiImpl({InstanceManager? instanceManager}) {
+    this.instanceManager = instanceManager ?? InstanceManager.instance;
+  }
+
+  /// Maintains instances stored to communicate with native language objects.
+  late final InstanceManager instanceManager;
+
+  @override
+  void didFinishNavigation(
+    int functionInstanceId,
+    int webViewInstanceId,
+    String? url,
+  ) {
+    final void Function(
+      WKWebView webView,
+      String? url,
+    ) function =
+        instanceManager.getInstance(functionInstanceId)! as void Function(
+      WKWebView webView,
+      String? url,
```

Comment:
Seems worth typedefing this instead of repeating the signature twice.

Reply:
I actually realized that `InstanceManager.getInstance` actually uses generics to make this casting easier. So, I didn't actually need the `as ....`.